### ISSUE TITLE
Add diagnostic check for Thanos blocks API

### ISSUE THREAD

**Issue by michaelmdresser** [kubecost/cost-analyzer-helm-chart#1606]
**What problem are you trying to solve?**

While testing 1.96.0-rc.0, I ran into a problem where Thanos' credentials were messed up, meaning it was failing to write data to S3. It is frustrating that Kubecost reports successful ETL builds and mostly green diagnostics in this situation.

![](https://user-images.githubusercontent.com/17585786/183135101-62d97977-e952-4150-9e3f-8d3b5d3605f1.png)

![](https://user-images.githubusercontent.com/17585786/183135033-5b48eb92-ffc0-4fc7-ab9b-c8a6372e0618.png)

The problem surfaced when I saw valid rebuilds but no data (see https://github.com/kubecost/Release-Testing/issues/29, see issue for valid build logs as well). We dug in:

1. Thanos query frontend was only reporting 1 store:
  `kubectl port-forward -n s3-backup svc/s3-backup-thanos-query-frontend-http 10902`
  ![](https://user-images.githubusercontent.com/17585786/183134663-a5f7bbcd-2df8-4359-95f8-0bdc9833f044.png)
1. Thanos store was reporting no blocks:
  `kubectl port-forward -n s3-backup svc/s3-backup-thanos-store-http 10902`
  ![](https://user-images.githubusercontent.com/17585786/183134744-18d0e5bf-d825-4782-abb8-313c0b7527e9.png)

```
→ curl 'http://localhost:10902/api/v1/blocks?view=loaded'
{"status":"success","data":{"label":"","blocks":[],"refreshedAt":"2022-08-05T18:01:18.569858736Z","err":null}}
```

1. The underlying problem was found by investigating Thanos sidecar logs

```
→ k logs -n s3-backup -l app=prometheus -l component=server -c thanos-sidecar --tail=50
...
level=info ts=2022-08-05T18:05:46.689831117Z caller=shipper.go:338 msg="upload new block" id=01G9MNXQ9N91X4T93NYWQEBQCZ
level=warn ts=2022-08-05T18:05:46.749262096Z caller=sidecar.go:339 err="upload 01G9MNXQ9N91X4T93NYWQEBQCZ: upload debug meta file: upload s3 object: Access Denied" uploaded=0
```

**Describe the solution you'd like**

I would like Kubecost to show a diagnostic message when Thanos is enabled but the store has no blocks.

For the store blocks, `/api/v1/blocks?view=loaded` seems like a good API to use because that's what the Thanos store frontend uses.

**Describe alternatives you've considered**

We could alternatively display a diagnostic when the Thanos sidecar has error/warn logs recently.

**How would users interact with this feature?**

On the settings and/or diagnostic page.

┆Issue is synchronized with this [Jira Task](https://kubecost.atlassian.net/browse/GIB-146) by [Unito](https://www.unito.io)
